packages/components/src/list/experimental-list.tsx

```diff
@@ -0,0 +1,53 @@
+/**
+ * External dependencies
+ */
+import { Children, cloneElement, isValidElement } from '@wordpress/element';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+
+type ListProps = {
+	component?: string;
+} & React.HTMLAttributes< HTMLElement >;
+
+export const ExperimentalList: React.FC< ListProps > = ( {
+	children,
+	component = 'nav',
+	// Allow passing any other property overrides that are legal on an HTML element
+	...otherProps
+} ) => {
+	return (
+		<TransitionGroup
+			component={ component }
+			className="woocommerce-list"
+			{ ...otherProps }
+		>
+			{ /* Wrapping all children in a CSS Transition means no invalid props are passed to children and that anything can be animated. */ }
+			{
+				Children.map( children, ( child ) => {
+					if ( isValidElement( child ) ) {
```

Comment:
Would you also want to check if the child is a ExperimentalListItem? Given `isValidElement` just checks if it is an element.

Reply:
@louwie17 not in this case. any child is legal and we have to avoid a div getting injected with the props from TransitionGroup